src/pyscipopt/scip.pyx

```diff
@@ -1276,9 +1276,8 @@ cdef class Model:
                 transcons = <Constraint>self.getTransformedCons(cons)
             else:
                 transcons = cons
-            if _nvars > 1:
-                dual = SCIPgetDualsolLinear(self._scip, transcons.cons)
-            else:
+            dual = SCIPgetDualsolLinear(self._scip, transcons.cons)
+            if dual == 0.0 and _nvars == 1:
```

Comment:
I don't quite get the change. Looks very cosmetic to me. Also, why would we trust `SCIPgetDualsolLinear` if it returns nonzero when the constraint contains only one variable?

Reply:
Apparently, sometimes a bound constraint appears as a constraint in the LP and we actually get a proper dual value while at other times, it gets converted into variable bounds and we need to ask for the reduced costs. In this case, the `getDualsolLinear()` call returns 0. Yeah, it is a workaround, but until we don't properly implement this in SCIP itself, we have to deal with it somehow.